src/main.nr

```diff
@@ -7,14 +8,24 @@ fn main(value: u32, hash_value: pub u32){
 }
 
 fn calculate_hash(num: u32) -> u32 {
-    
+    let mut hash_value = 0x811C9DC5;
+    let bytes = (num as Field).to_be_bytes(4);
+
+    for i in 0..32 {
+        let byte = bytes[i] as u64;
+        if byte != 0x00 {
+            let product = ((hash_value * PRIME) as u64) % SIZE;
```

Comment:
When the value of `hash_value * PRIME` (in u32) goes beyond the limit, Noir handles the mod operation. You can skip the % with SIZE here if you are doing it without casting to u64